iree-requirements-ci.txt

```diff
@@ -4,7 +4,10 @@
 # forgiving on the exact version.
 
 # Uncomment to select a nightly version.
---find-links https://iree.dev/pip-release-links.html
+# --find-links https://iree.dev/pip-release-links.html
 
-iree-compiler==20241107.1071
-iree_runtime==20241107.1071
+# Uncomment to skip versions from PyPI (so _only_ nightly versions).
+# --no-index
+
+iree-base-compiler<=2.9.0
```

Comment:
CI failure logs: https://github.com/iree-org/iree-turbine/actions/runs/11750953131/job/32740143979?pr=258

Running this locally, with extra printf debugging:
```
pytest . --log-cli-level=debug -k compiled_exported
```

Prior release:
```
----------------------------------------------------------------------- Captured stdout call ------------------------------------------------------------------------
materialized_global.global_op:
"util.global"() <{initial_value = dense_resource<_buffers.buf> : tensor<1xf32>, is_mutable, sym_name = "_buffers.buf", sym_visibility = "private", type = tensor<1xf32>}> : () -> ()
<class 'iree.compiler.dialects._util_ops_gen.GlobalOp'>
-------------------------------------------------------
```

Latest release:
```
----------------------------------------------------------------------- Captured stdout call ------------------------------------------------------------------------
materialized_global.global_op:
"util.global"() <{initial_value = dense_resource<_buffers.buf> : tensor<1xf32>, is_mutable, sym_name = "_buffers.buf", sym_visibility = "private", type = tensor<1xf32>}> : () -> ()
<class 'iree.compiler._mlir_libs._mlir.ir.Operation'>
------------------------------------------------------------------------- Captured log call -------------------------------------------------------------------------
```

Reply:
Needed to adapt to https://github.com/llvm/llvm-project/commit/00a93e6207ac053e7a42368a52cf51a744ecd139.